src/Google.Api.Gax/CallSettings.cs

```diff
@@ -62,6 +73,65 @@ public sealed class CallSettings
             WriteOptions = WriteOptions,
             PropagationToken = PropagationToken,
             Credentials = Credentials,
+            RetrySettings = RetrySettings?.Clone(),
         };
+
+        // Merge other into this.
+        // Other is unchanged. Returns this
+        // Internal? Or Public?
+        public CallSettings Merge(CallSettings other)
+        {
+            if (other == null)
+            {
+                return this;
+            }
+            if (other.Headers != null)
+            {
+                Headers = other.Headers.Clone();
+            }
+            if (other.Expiration != null)
+            {
+                Expiration = other.Expiration;
+            }
+            if (other.CancellationToken != null)
+            {
+                CancellationToken = other.CancellationToken;
+            }
+            if (other.WriteOptions != null)
+            {
+                WriteOptions = other.WriteOptions;
+            }
+            if (other.PropagationToken != null)
+            {
+                PropagationToken = other.PropagationToken;
+            }
+            if (other.Credentials != null)
+            {
+                Credentials = other.Credentials;
+            }
+            if (other.RetrySettings != null)
+            {
+                RetrySettings = other.RetrySettings;
+            }
+            return this;
+        }
+
+        internal CallOptions ToCallOptions(IClock clock) => new CallOptions(
+            headers: Headers,
+            deadline: Expiration.CalculateDeadline(clock),
+            cancellationToken: CancellationToken ?? default(CancellationToken),
+            writeOptions: WriteOptions,
+            propagationToken: PropagationToken,
+            credentials: Credentials);
+
+        internal CallSettings AddUserAgent(string userAgent)
```

Comment:
I can't remember where this gets called from now, but it's likely to be relevant with the header overwriting.